src/Jackett.Common/Indexers/FileList.cs

```diff
@@ -99,109 +100,169 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
                 response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
             }
 
-            var results = response.Content;
+            return response.Content;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var pageCount = 0;
+            var firstResponse = await callProviderAsync(query, 0);
+
+            pageCount = getPageCount(firstResponse);
+            transformProviderRequest(query, firstResponse, releases);
+            //Wait
+            Thread.Sleep(50);
+
+            //Process the rest of the pages
+            if (pageCount > 0 && query.GetQueryString() != "")
+            {
+                for (int i = 1; i < pageCount; i++)
+                {
+                    var resp = await callProviderAsync(query, i);
+                    transformProviderRequest(query, resp, releases);
+                    //Wait between requests
+                    Thread.Sleep(50);
+                }
+
+            }
+
+            return releases;
+        }
+
+        private Int16 getPageCount(String html)
+        {
+            try
+            {
+                var parser = new HtmlParser();
+                var dom = parser.ParseDocument(html);
+                var pageCount = dom.QuerySelectorAll(".pager > span > a > font").Last().InnerHtml;
+                return Int16.Parse(pageCount);
+            }
+            catch (Exception ex)
+            {
+                OnParseError(html, ex);
+            }
+            return 0;
+        }
+
+        private void transformProviderRequest(TorznabQuery query, String results, List<ReleaseInfo> releases)
+        {
             try
             {
                 var parser = new HtmlParser();
                 var dom = parser.ParseDocument(results);
-                var globalFreeLeech = dom.QuerySelectorAll("div.globalFreeLeech").Any();
-                var rows = dom.QuerySelectorAll(".torrentrow");
-                foreach (var row in rows)
-                {
-                    var release = new ReleaseInfo();
+                parseResults(dom, query, releases);
+            }
+            catch (Exception ex)
+            {
+                OnParseError(results, ex);
+            }
+        }
 
-                    var qTitleLink = row.QuerySelector(".torrenttable:nth-of-type(2) a");
-                    release.Title = row.QuerySelector(".torrenttable:nth-of-type(2) b").TextContent;
-                    var longtitle = row.QuerySelector(".torrenttable:nth-of-type(2) a[title]").GetAttribute("title");
-                    if (!string.IsNullOrEmpty(longtitle) && !longtitle.Contains("<")) // releases with cover image have no full title
-                        release.Title = longtitle;
+        private void parseResults(AngleSharp.Html.Dom.IHtmlDocument dom, TorznabQuery query, List<ReleaseInfo> releases)
```

Comment:
Transforming a reference type passed as a regular parameter is discouraged. There are two main ways in C# to indicate that a function will be modifying `releases`
```C#
private void parseResults(AngleSharp.Html.Dom.IHtmlDocument dom, TorznabQuery query, ref List<ReleaseInfo> releases)
```
or
```C#
private List<ReleaseInfo> parseResults(AngleSharp.Html.Dom.IHtmlDocument dom, TorznabQuery query)
```

We prefer the second option for clarity. One further step to avoid allocating a temporary List that gets returned would be to return `IEnumerable<ReleaseInfo>` and use `yield return release;` everywhere your existing code has a call to `releases.Add()`